plugins/MakerbotWriter/MakerbotWriter.py

```diff
@@ -0,0 +1,278 @@
+                comment="Makerbot Toolpath Package",
+                suffixes=["makerbot"]
+            )
+        )
+
+    _PNG_FORMATS = [
+        {"prefix": "isometric_thumbnail", "width": 120, "height": 120},
+        {"prefix": "isometric_thumbnail", "width": 320, "height": 320},
+        {"prefix": "isometric_thumbnail", "width": 640, "height": 640},
+        {"prefix": "thumbnail", "width": 140, "height": 106},
+        {"prefix": "thumbnail", "width": 212, "height": 300},
+        {"prefix": "thumbnail", "width": 960, "height": 1460},
+        {"prefix": "thumbnail", "width": 90, "height": 90},
+    ]
+    _META_VERSION = "3.0.0"
+    _PRINT_NAME_MAP = {
+        "Makerbot Method": "fire_e",
+        "Makerbot Method X": "lava_f",
+        "Makerbot Method XL": "magma_10",
+    }
+    _EXTRUDER_NAME_MAP = {
+        "1XA": "mk14_hot",
+        "2XA": "mk14_hot_s",
+        "1C": "mk14_c",
+        "1A": "mk14",
+        "2A": "mk14_s",
+    }
+    _MATERIAL_MAP = {
+        '2780b345-577b-4a24-a2c5-12e6aad3e690': 'abs',
+        '88c8919c-6a09-471a-b7b6-e801263d862d': 'abs-wss1',
+        '416eead4-0d8e-4f0b-8bfc-a91a519befa5': 'asa',
+        '85bbae0e-938d-46fb-989f-c9b3689dc4f0': 'nylon-cf',
+        '283d439a-3490-4481-920c-c51d8cdecf9c': 'nylon',
+        '62414577-94d1-490d-b1e4-7ef3ec40db02': 'pc',
+        '69386c85-5b6c-421a-bec5-aeb1fb33f060': 'pet',  # PETG
+        '0ff92885-617b-4144-a03c-9989872454bc': 'pla',
+        'a4255da2-cb2a-4042-be49-4a83957a2f9a': 'pva',
+        'a140ef8f-4f26-4e73-abe0-cfc29d6d1024': 'wss1',
+        '77873465-83a9-4283-bc44-4e542b8eb3eb': 'sr30',
+        '96fca5d9-0371-4516-9e96-8e8182677f3c': 'im-pla',
+        '19baa6a9-94ff-478b-b4a1-8157b74358d2': 'tpu',
+    }
+
+    # must be called from the main thread because of OpenGL
+    @staticmethod
+    @call_on_qt_thread
+    def _createThumbnail(width: int, height: int) -> Optional[QBuffer]:
+        if not CuraApplication.getInstance().isVisible:
+            Logger.warning("Can't create snapshot when renderer not initialized.")
+            return
+        try:
+            snapshot = Snapshot.isometric_snapshot(width, height)
+        except:
+            Logger.logException("w", "Failed to create snapshot image")
+            return
+
+        thumbnail_buffer = QBuffer()
+        thumbnail_buffer.open(QBuffer.OpenModeFlag.ReadWrite)
+
+        snapshot.save(thumbnail_buffer, "PNG")
+
+        return thumbnail_buffer
+
+    def write(self, stream: BufferedIOBase, nodes: List[SceneNode], mode=MeshWriter.OutputMode.BinaryMode) -> bool:
+        if mode != MeshWriter.OutputMode.BinaryMode:
+            Logger.log("e", "MakerbotWriter does not support text mode.")
+            self.setInformation(catalog.i18nc("@error:not supported", "MakerbotWriter does not support text mode."))
+            return False
+
+        # The GCodeWriter plugin is bundled, so it must at least exist. (What happens if people disable that plugin?)
```

Comment:
GCode Writer plugin does not show up in my plugins list, so it seems impossible to disable it..

Reply:
@casper will add this, good catch

Reply:
The gcode writer is in the required plugins so it is always available. Updated comment to reflect this behavior: https://github.com/Ultimaker/Cura/pull/17147/commits/6a07c346d6a6d6299756e36a7c9b3086ed60944e